### ISSUE TITLE
Support spacemice

### ISSUE THREAD

**Issue by janEntikan** [zturtleman/mm3d#187]
MM3D's workflow lends itself quite perfectly for use with a 3d mouse like the spacemouse/spacenavigator/cadmouse/etc to allow for 6 degrees of freedom transformation and navigation.

**Comment by zturtleman** [comment]
I don't plan to work on it. Adding support for these devices is non-trivial. It seems difficult to integrate support for, the 3Dconnexion Spacemouse devices are cost prohibitive, and there is probably a fair amount of design consideration to integrate it well.

The [official 3Dconnexion SDK](https://3dconnexion.com/us/software-developer-program/) for Windows/macOS seems to be under an license that cannot be used with Maverick. (I don't know what the specific terms are as I didn't register for access.) There is an unofficial Linux [spacenav driver/SDK](https://spacenav.sourceforge.net) could be used on Linux.

**Comment by janEntikan** [comment]
Indeed, I am using the unofficial linux driver. I'm willing to look into this a bit later. Is this perhaps better suited as a plugin?

**Comment by zturtleman** [comment]
It would probably need to affect the built-in tools though? I guess it might be possible to add new "3D move", "3D extrude", etc commands as a plugin. If it's optional at compile time and not too intrusive, it can probably go in the main repo.
